src/OpenTelemetry.Api/AssemblyInfo.cs

```diff
@@ -19,6 +19,7 @@
 [assembly: InternalsVisibleTo("OpenTelemetry" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Api.ProviderBuilderExtensions" + AssemblyInfo.PublicKey)]
 [assembly: InternalsVisibleTo("OpenTelemetry.Tests" + AssemblyInfo.PublicKey)]
```

Comment:
I guess we should check if it would be possible to get rid of exposing internals to `OpenTelemetry.Tests` if we create a separate test project for OpenTelemetry.Api.

Reply:
Yes, I've already scoped this out. :)
This first PR creates the empty project. 
I've got another branch ready to move over the tests.